### PR TITLE
Support re-packing NuGet content

### PR DESCRIPTION
**Pull request by jaredpar**
This adds support for re-packing nupkg files with signed binaries. Can't actually sign the nupkg themselves but the tool can now repack the contents of the nupkg with signed content. Exactly as we do for VSIX today.

This should allow for some simplification of our build logic. Can now build everything, then sign. Where as today we have to build, sign, pack then build some more.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @tmat, @tannergooding

File: src/SignTool/SignTool/SignTool.csproj [hunk 1]
```diff
@@ -1,42 +1,26 @@
...
-    <Compile Include="SignTool.SignToolBase.cs" />
-    <Compile Include="SignTool.TestSignTool.cs" />
-    <Compile Include="SignToolArgs.cs" />
-    <Compile Include="VsixData.cs" />
-  </ItemGroup>
-  <ItemGroup>
     <Reference Include="System" />
     <Reference Include="WindowsBase" />
+    <PackageReference Include="Microsoft.Extensions.FileSystemGlobbing">
+      <Version>2.0.0</Version>
```
Comment (tmat) [inline src/SignTool/SignTool/SignTool.csproj]: ><Version>2.0.0</Version> [](start = 6, length = 24)

Nit: attribute?

File: src/SignTool/SignTool/VsixData.cs [hunk 2]
```diff
@@ -9,33 +9,36 @@
 
 namespace SignTool
 {
-    internal sealed class VsixData
+    /// <summary>
+    /// Data for a zip container. Can refer to any zip format such as VSIX or nupkg
+    /// </summary>
+    internal sealed class ZipData
```
Comment (tmat) [inline src/SignTool/SignTool/VsixData.cs]: >ZipData [](start = 26, length = 7)

Rename the file to "ZipData"

File: src/SignTool/SignTool/VsixData.cs [hunk 3]
```diff
@@ -47,17 +50,19 @@ internal VsixData(FileName name, ImmutableArray<VsixPart> nestedBinaryParts, Imm
         }
     }
 
-    internal struct VsixPart
+    internal struct ZipPart
```
Comment (tmat) [inline src/SignTool/SignTool/VsixData.cs]: >ZipPart [](start = 20, length = 7)

Move to a separate file.

File: src/SignTool/SignTool/SignTool.RealSignTool.cs [hunk 4]
```diff
@@ -79,7 +79,7 @@ public override void RemovePublicSign(string assemblyPath)
                     }
 
                     stream.Position = peReader.PEHeaders.CorHeaderStartOffset + OffsetFromStartOfCorHeaderToFlags;
-                    writer.Write((UInt32)(peReader.PEHeaders.CorHeader.Flags | CorFlags.StrongNameSigned));
+                    writer.Write((UInt32)(peReader.PEHeaders.CorHeader.Flags & ~CorFlags.StrongNameSigned));
```
Comment (tmat) [inline src/SignTool/SignTool/SignTool.RealSignTool.cs]: >& ~ [](start = 77, length = 3)

Was this always wrong?

**Comment by tmat** [comment]
👍

File: src/SignTool/SignTool/SignTool.RealSignTool.cs (hunk 4 above)
Comment (jaredpar) [reply src/SignTool/SignTool/SignTool.RealSignTool.cs]: Yes. @nguerrera pointed this out to me a while ago.

File: src/SignTool/SignTool/VsixData.cs (hunk 3 above)
Comment (jaredpar) [reply src/SignTool/SignTool/VsixData.cs]: K

File: src/SignTool/SignTool/VsixData.cs (hunk 2 above)
Comment (jaredpar) [reply src/SignTool/SignTool/VsixData.cs]: K

File: src/SignTool/SignTool/SignTool.csproj (hunk 1 above)
Comment (jaredpar) [reply src/SignTool/SignTool/SignTool.csproj]: Fixed

**Pull request merged**